yt_dlp/extractor/laxarxames.py

```diff
@@ -0,0 +1,90 @@
+from .common import InfoExtractor
+
+
+class LaXarxaMesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?laxarxames\.cat/(?:[^/]+/)*?(player|movie-details)/(?P<id>[0-9]+)'
+    _NETRC_MACHINE = 'laxarxames'
+    _IS_LOGGED_IN = False
+    _LOGIN_URL = 'https://www.laxarxames.cat/login'
+    _TESTS = [{
+        'url': 'https://www.laxarxames.cat/player/3459421',
+        'md5': '0966f46c34275934c19af78f3df6e2bc',
+        'info_dict': {
+            'id': '3459421',
+            'ext': 'mp4',
+            'title': 'Resum | UA Horta — UD Viladecans',
+            'type': 'video/mp4',
+        },
+        'skip': 'Requires login',
+    }]
+
+    def _perform_login(self, username, password):
+        login = self._download_json(
+            'https://api.laxarxames.cat/Authorization/SignIn', None, note='Logging in', headers={
+                'X-Tenantorigin': 'https://laxarxames.cat',
+                'Content-Type': 'application/json',
+                'Accept': 'application/json, text/plain, */*',
+                'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko)',
+                'Origin': 'https://www.laxarxames.cat',
+            }, data=b'{"Username":"%s","Password":"%s","Device":{"PlatformCode":"WEB","Name":"Mac OS ()"}}' % (username.encode(), password.encode())
+        )
+
+        if not login['AuthorizationToken']:
+            raise Exception('Login failed')
+        else:
+            self._set_cookie('www.laxarxames.cat', 'didomi_token', login['AuthorizationToken']['Token'])
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        authorization = self._get_cookies('https://www.laxarxames.cat/').get('didomi_token')
```

Comment:
doing it via cookie like this makes sense if the auth cookies are only scoped to `www.laxarxames.cat` and not to `api.laxarxames.cat`, so that `--cookies`/`--cookies-from-browser` can be handled equivalently to `-u`/`-p`

should do it like this though:
```py
        authorization = try_call(self._get_cookies(lambda: 'https://www.laxarxames.cat/')['didomi_token'].value)
        ```

Reply:
My suspicion however was that the required Bearer token is not actually stored in a cookie, but say uses localStorage.
I just created an account to test this, and the `didomi_token` cookie does in fact not match the Bearer token used for API requests. The bearer token is instead stored a JSON string in the localStorage with the key `session`.

Reply:
ah, in that case you're right we should definitely not use the cookie workaround

Reply:
Thank you to you both for your help!
Is there any way yt-dlp can access localStorage and get the token?

Reply:
Currently the browsers localStorage can't be accessed from yt-dlp.
imo you should write the token to a class variable.